java/gandiva/src/test/java/org/apache/arrow/gandiva/evaluator/FilterTest.java

```diff
@@ -44,6 +49,121 @@
     return actual;
   }
 
+  private Charset utf8Charset = Charset.forName("UTF-8");
+  private Charset utf16Charset = Charset.forName("UTF-16");
+
+  List<ArrowBuf> varBufs(String[] strings, Charset charset) {
```

Comment:
Could you use ValueVectorDataPopulator instead and extract the buffers from there instead of hand creating the buffers?